File: Creeds/app/src/main/java/com/nonlinearfruit/creeds/main/MainDatabase.java
package com.nonlinearfruit.creeds.main;

import com.nonlinearfruit.creeds.R;
import com.nonlinearfruit.creeds.canon.CanonActivity;
import com.nonlinearfruit.creeds.catechism.CatechismActivity;
import com.nonlinearfruit.creeds.creed.CreedActivity;
import com.nonlinearfruit.creeds.creed.CreedLookup;
import com.nonlinearfruit.creeds.henryscatechism.HenrysCatechismActivity;
import com.nonlinearfruit.creeds.main.models.MainMenuItem;
import com.nonlinearfruit.creeds.confession.ConfessionActivity;

import java.util.ArrayList;
import java.util.List;

public class MainDatabase {

    private List<MainMenuItem> items = new ArrayList<MainMenuItem>(){{
        //add(new MainMenuItem(){{
        //    CreedTitle = "First Catechism";
        //    CreedYear = 1996;
        //    CreedOrigin = "Copyright 2003 - Great Commission Publications, Inc";
        //    IntentClass = CatechismActivity.class;
        //    JsonFileId = R.raw.first_catechism;
        //}});
        add(new MainMenuItem(){{
            CreedTitle = "Chicago Statement on Biblical Inerrancy";
            CreedYear = 1978;
            CreedOrigin = "Copyright 1978 - International Council on Biblical Inerrancy";
            IntentClass = CanonActivity.class;
            JsonFileId = R.raw.chicago_statement_on_biblical_inerrancy;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Puritan Catechism";
            CreedYear = 1855;
            CreedOrigin = "";
            IntentClass = CatechismActivity.class;
            JsonFileId = R.raw.puritan_catechism;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Matthew Henry's Scripture Catechism";
            CreedYear = 1714;
            CreedOrigin = "";
            IntentClass = HenrysCatechismActivity.class;
            JsonFileId = R.raw.matthew_henrys_scripture_catechism;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "London Baptist 1689 Confession of Faith";
            CreedYear = 1689;
            CreedOrigin = "";
            IntentClass = ConfessionActivity.class;
            JsonFileId = R.raw.london_baptist_1689;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Keach's Catechism";
            CreedYear = 1677;
            CreedOrigin = "";
            IntentClass = CatechismActivity.class;
            JsonFileId = R.raw.keachs_catechism;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Westminster Shorter Catechism";
            CreedYear = 1647;
            CreedOrigin = "";
            IntentClass = CatechismActivity.class;
            JsonFileId = R.raw.westminster_shorter_catechism;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Westminster Larger Catechism";
            CreedYear = 1647;
            CreedOrigin = "";
            IntentClass = CatechismActivity.class;
            JsonFileId = R.raw.westminster_larger_catechism;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Westminster Confession of Faith";
            CreedYear = 1647;
            CreedOrigin = "";
            IntentClass = ConfessionActivity.class;
            JsonFileId = R.raw.westminster_confession_of_faith;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Canons of Dort";
            CreedYear = 1619;
            CreedOrigin = "";
            IntentClass = ConfessionActivity.class;
            JsonFileId = R.raw.canons_of_dort;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Heidelberg Catechism";
            CreedYear = 1563;
            CreedOrigin = "";
            IntentClass = CatechismActivity.class;
            JsonFileId = R.raw.heidelberg_catechism;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Second Helvetic Confession";
            CreedYear = 1562;
            CreedOrigin = "";
            IntentClass = ConfessionActivity.class;
            JsonFileId = R.raw.second_helvetic_confession;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Belgic Confession of Faith";
            CreedYear = 1561;
            CreedOrigin = "";
            IntentClass = CanonActivity.class;
            JsonFileId = R.raw.belgic_confession_of_faith;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Scots Confession";
            CreedYear = 1560;
            CreedOrigin = "";
            IntentClass = CanonActivity.class;
            JsonFileId = R.raw.scots_confession;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "French Confession of Faith";
            CreedYear = 1559;
            CreedOrigin = "";
            IntentClass = CanonActivity.class;
            JsonFileId = R.raw.french_confession_of_faith;
        }});
        //add(new MainMenuItem(){{
        //    CreedTitle = "Augsburg Confession";
        //    CreedYear = 1530;
        //    CreedOrigin = "";
        //    IntentClass = ConfessionActivity.class;
        //    JsonFileId = R.raw.augsburg_confession;
        //}});
        add(new MainMenuItem(){{
            CreedTitle = "Zwingli's 67 Articles";
            CreedYear = 1523;
            CreedOrigin = "";
            IntentClass = CanonActivity.class;
            JsonFileId = R.raw.zwinglis_67_articles;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Athanasian Creed";
            CreedYear = 800;
            CreedOrigin = "The oldest surviving manuscripts of the Athanasian Creed date from the late 8th century";
            IntentClass = CreedActivity.class;
            JsonFileId = CreedLookup.AthanasianCreed.Id;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Apostles' Creed";
            CreedYear = 710;
            CreedOrigin = "De singulis libris canonicis scarapsus ('Excerpt from Individual Canonical Books') of St. Pirminius (Migne, Patrologia Latina 89, 1029 ff.), written between 710 and 714.";
            IntentClass = CreedActivity.class;
            JsonFileId = CreedLookup.ApostlesCreed.Id;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Chalcedonian Definition";
            CreedYear = 451;
            CreedOrigin = "Adopted at the Council of Chalcedon in AD 451";
            IntentClass = CreedActivity.class;
            JsonFileId = CreedLookup.ChalcedonianDefinition.Id;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Nicene Creed";
            CreedYear = 381;
            CreedOrigin = "Adopted at the Second Ecumenical Council held in Constantinople in 381 as a modification of the original Nicene Creed of 325";
            IntentClass = CreedActivity.class;
            JsonFileId = CreedLookup.NiceneCreed.Id;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Confession of Peter";
            CreedYear = 30;
            CreedOrigin = "Matthew 16:16";
            IntentClass = CreedActivity.class;
            JsonFileId = CreedLookup.ConfessionOfPeter.Id;
        }});
        add(new MainMenuItem(){{
            CreedTitle = "Shema Yisrael";
            CreedYear = -1500;
            CreedOrigin = "Deuteronomy 6:4";
            IntentClass = CreedActivity.class;
            JsonFileId = CreedLookup.ShemaYisrael.Id;
        }});
    }};

    public List<MainMenuItem> getMainMenuItems() {
        return items;
    }
}

File: Creeds/app/src/main/java/com/nonlinearfruit/creeds/main/MainDatabase.kt
package com.nonlinearfruit.creeds.main

import com.nonlinearfruit.creeds.R
import com.nonlinearfruit.creeds.canon.CanonActivity
import com.nonlinearfruit.creeds.catechism.CatechismActivity
import com.nonlinearfruit.creeds.creed.CreedActivity
import com.nonlinearfruit.creeds.henryscatechism.HenrysCatechismActivity
import com.nonlinearfruit.creeds.main.models.MainMenuItem
import com.nonlinearfruit.creeds.confession.ConfessionActivity

class MainDatabase {

        val mainMenuItems: List<MainMenuItem> = listOf(
MainMenuItem( "Abstract of Principles", 1858, "", CanonActivity::class.java, R.raw.abstract_of_principles),
MainMenuItem( "Apostles' Creed", 710, "", CreedActivity::class.java, R.raw.apostles_creed),
MainMenuItem( "Athanasian Creed", 800, "", CreedActivity::class.java, R.raw.athanasian_creed),
MainMenuItem( "Belgic Confession", 1561, "", CanonActivity::class.java, R.raw.belgic_confession_of_faith),
MainMenuItem( "Canons of Dort", 1619, "", ConfessionActivity::class.java, R.raw.canons_of_dort),
MainMenuItem( "Catechism For Young Children", 1840, "", CatechismActivity::class.java, R.raw.catechism_for_young_children),
MainMenuItem( "Chalcedonian Definition", 451, "", CreedActivity::class.java, R.raw.chalcedonian_definition),
MainMenuItem( "Chicago Statement on Biblical Inerrancy", 1978, "", CanonActivity::class.java, R.raw.chicago_statement_on_biblical_inerrancy),
MainMenuItem( "Christ Hymn of Colossians", 60, "", CreedActivity::class.java, R.raw.christ_hymn_of_colossians),
MainMenuItem( "Christ Hymn of Philippians", 60, "", CreedActivity::class.java, R.raw.christ_hymn_of_philippians),
MainMenuItem( "Confession of Peter", 30, "", CreedActivity::class.java, R.raw.confession_of_peter),
MainMenuItem( "Consensus Tigurinus", 1549, "", CanonActivity::class.java, R.raw.consensus_tigurinus),
MainMenuItem( "Council of Orange", 529, "", CanonActivity::class.java, R.raw.council_of_orange),
MainMenuItem( "First Catechism", 1996, "", CatechismActivity::class.java, R.raw.first_catechism),
MainMenuItem( "First Confession of Basel", 1534, "", CanonActivity::class.java, R.raw.first_confession_of_basel),
MainMenuItem( "First Helvetic Confession", 1536, "", CanonActivity::class.java, R.raw.first_helvetic_confession),
MainMenuItem( "French Confession of Faith", 1559, "", CanonActivity::class.java, R.raw.french_confession_of_faith),
MainMenuItem( "Heidelberg Catechism", 1563, "", CatechismActivity::class.java, R.raw.heidelberg_catechism),
MainMenuItem( "Helvetic Consensus", 1675, "", CanonActivity::class.java, R.raw.helvetic_consensus),
MainMenuItem( "Irenaeus' Rule of Faith", 180, "", CreedActivity::class.java, R.raw.irenaeus_rule_of_faith),
MainMenuItem( "Keach's Catechism", 1693, "", CatechismActivity::class.java, R.raw.keachs_catechism),
MainMenuItem( "1689 London Baptist Confession", 1677, "", ConfessionActivity::class.java, R.raw.london_baptist_1689),
MainMenuItem( "Matthew Henry's Scripture Catechism", 1703, "", HenrysCatechismActivity::class.java, R.raw.matthew_henrys_scripture_catechism),
MainMenuItem( "Nicene Creed", 381, "", CreedActivity::class.java, R.raw.nicene_creed),
MainMenuItem( "Puritan Catechism", 1855, "", CatechismActivity::class.java, R.raw.puritan_catechism),
MainMenuItem( "Scots Confession", 1560, "", CanonActivity::class.java, R.raw.scots_confession),
MainMenuItem( "Second Helvetic Confession", 1562, "", ConfessionActivity::class.java, R.raw.second_helvetic_confession),
MainMenuItem( "Shema Yisrael", -1500, "", CreedActivity::class.java, R.raw.shema_yisrael),
MainMenuItem( "Ten Theses of Berne", 1528, "", CanonActivity::class.java, R.raw.ten_theses_of_berne),
MainMenuItem( "Tetrapolitan Confession", 1530, "", CanonActivity::class.java, R.raw.tetrapolitan_confession),
MainMenuItem( "Waldensian Confession", 1120, "", CanonActivity::class.java, R.raw.waldensian_confession),
MainMenuItem( "Westminster Confession of Faith", 1646, "", ConfessionActivity::class.java, R.raw.westminster_confession_of_faith),
MainMenuItem( "Westminster Larger Catechism", 1647, "", CatechismActivity::class.java, R.raw.westminster_larger_catechism),
MainMenuItem( "Westminster Shorter Catechism", 1647, "", CatechismActivity::class.java, R.raw.westminster_shorter_catechism),
MainMenuItem( "Zwingli's 67 Articles", 1523, "", CanonActivity::class.java, R.raw.zwinglis_67_articles)
)}
